hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -150,6 +157,37 @@ public ClientEngineImpl(Node node) {
         this.partitionListenerService = new ClientPartitionListenerService(nodeEngine);
     }
 
+    @Override
+    public void probeIn(ProbingCycle cycle) {
+        cycle.probe("client.endpoint", endpointManager);
+        if (!cycle.isProbed(ProbeLevel.INFO)) {
+            return;
+        }
+        Collection<ClientEndpoint> endpoints = endpointManager.getEndpoints();
+        if (!endpoints.isEmpty()) {
+            for (ClientEndpoint endpoint : endpoints) {
+                Tags tags = cycle.openContext().tag(TAG_TYPE, "client");
+                if (endpoint.isAlive()) {
+                    tags.tag(TAG_INSTANCE, endpoint.getUuid());
+                    cycle.probe(endpoint);
+                    String type = endpoint.getClientType().name().toLowerCase();
+                    InetSocketAddress remote = endpoint.getConnection().getRemoteSocketAddress();
+                    String address = remote == null ? "?" : remote.getAddress().getHostAddress() + ":" + remote.getPort();
```

Comment:
This is going to generate quite a lot of litter.

Reply:
Good point. We can avoid these.